### PR TITLE
add Lessgas

### PR DESCRIPTION
**Pull request by IvanMacaron**
##### Name (to be shown on DefiLlama):
Lessgas

##### Twitter Link:
https://twitter.com/Less_Gas

##### Website Link:
https://lessgas.xyz

##### Logo (High resolution, will be shown with rounded borders):
icon_logo

##### Current TVL:
Map Protocol： 5.42 M USDT

##### Chain:
 Map Protocol

##### Short Description (to be shown on DefiLlama):
Bitcoin Layer2 inscription platform with ultra-low gas fees,zero failures,and zero congestion

##### Token address and ticker if any:
$LSGS Map Protocol 0x756af1d3810a01d3292fad62f295bbcc6c200aea
$LSGS Bitcore 4667a31423135f0fe7e8ce3cd8aa60cb493e9e201e8135785a1c7fcda7a52c0ci0

##### Category (full list at https://defillama.com/categories) *Please choose only one:
Launchpad

##### forkedFrom (Does your project originate from another project):
No, it is built on our own

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/lessgas exports TVL: 
        
 
 
    map-staking               1.67 M
    map                       5.41 M
    staking                   1.67 M
    
    total                    5.41 M

**Review by g1nt0ki — CHANGES_REQUESTED** [review]
Please refactor code to compute tvl on chain, in it's current state, I have no idea what is counted as tvl

**Comment by llamatester** [comment]
Error while running adapter at projects/lessgas: 
        
 
  
    
    
    Error: connect ECONNREFUSED 127.0.0.1:9912
    Truncated error stack:
        at AxiosError.from (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/axios/dist/node/axios.cjs:837:14)
        at RedirectableRequest.emit (node:events:517:28)
        at ClientRequest.emit (node:events:517:28)
        at Socket.socketErrorListener (node:_http_client:501:9)
        at Socket.emit (node:events:517:28)

**Review by g1nt0ki — CHANGES_REQUESTED** [review]
none of these tokens are in coingecko, we rely on their api for price, and why are you counting token supply as tvl? https://docs.llama.fi/list-your-project/what-to-include-as-tvl

**Comment by llamatester** [comment]
The adapter at projects/lessgas exports TVL: 
        
 
 
    map-staking               1.67 M
    map                       5.42 M
    staking                   1.67 M
    
    total                    5.42 M

**Comment by llamatester** [comment]
The adapter at projects/lessgas exports TVL: 
        
 
 
    map                       1.34 M
    map-staking               1.34 M
    staking                   1.34 M
    
    total                    1.34 M

**Comment by g1nt0ki** [comment]
> none of these tokens are in coingecko, we rely on their api for price, and why are you counting token supply as tvl? https://docs.llama.fi/list-your-project/what-to-include-as-tvl

@IvanMacaron you are double counting staking as tvl, we cant use your rest api for pricing tokens, for pulling token list it is fine (we cache api response on our end)

**Comment by IvanMacaron** [comment]
Hello sir, I am confused about the price. I do not know how to obtain the price from the API you provided. Can you please let me know specifically so that I can modify the relevant code and complete the data adaptation

**Comment by llamatester** [comment]
The adapter at projects/lessgas exports TVL: 
        
 
 
    map                       0
    map-staking               1.44 M
    staking                   1.44 M
    
    total                    0

File: projects/lessgas/index.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    try {
+      let { output } = await sdk.api.abi.call({
+        abi: "erc20:balanceOf",
+        target: item[1],
+        params: [item[0]],
+        chain: "map"
+        // block: chainBlocks["arbitrum"],
+      });
+      console.log(output);
+     total =  new BigNumber(total).plus(new BigNumber(output).div(10**18).multipliedBy(new BigNumber(item[2])));
```
Comment (g1nt0ki) [inline projects/lessgas/index.js:27]: I am talking about this line, item[2] here is the price right? we use our own api for pulling token price, for example xcCGF on moonbeam: https://coins.llama.fi/prices/current/moonbeam:0xffffffff44bd9d2ffee20b25d1cf9e78edb6eae3

We dont have a token list for map tokens, so currently it is returning 0

Comment (g1nt0ki) [reply projects/lessgas/index.js:27]: do you know of an official bridge, that I can use to map map tokens to brc20 on bitcoin, and/or what tokens on map  chain are listed on coingecko? without these info we cant price tokens on map

**Comment by g1nt0ki** [comment]
@IvanMacaron how are you pricing LSGS token? I dont find it on coingecko, so unsure how to price it

**Comment by IvanMacaron** [comment]
> @IvanMacaron how are you pricing LSGS token? I dont find it on coingecko, so unsure how to price it
Sir,
We are submitting our price information to Coinsecko，
I will inform you as soon as the submission is successful

**Comment by IvanMacaron** [comment]
> @IvanMacaron how are you pricing LSGS token? I dont find it on coingecko, so unsure how to price it

Sir,
Our prices can already be found on Coinsecko. How should I make adaptation modifications below?

**Comment by llamatester** [comment]
The adapter at projects/lessgas exports TVL: 
        
 
 
    map                       0
    map-staking               0
    staking                   0
    
    total                    0

**Comment by g1nt0ki** [comment]
@IvanMacaron sorry for the delay, I have added code to price LSGS tokens on our server repo. https://github.com/DefiLlama/defillama-server/commit/3b2a337f1ca4a5dd7d0386c8f121046156f3db37 so, LSGS will have price in 2-3 hours.

**Comment by IvanMacaron** [comment]
Sir，
Coinmarketcap supports the MAP Protocol main network, and other token prices on the MAP Protocol main network can be retrieved from Coinmarketcap. May I ask if it supports TVL for stacking other tokens included in Lessgas

**Pull request merged**